ext/hiredis_ext/connection.c

```diff
@@ -253,15 +264,83 @@ static VALUE connection_connect(int argc, VALUE *argv, VALUE self) {
             if (NUM2INT(arg_timeout) <= 0) {
                 rb_raise(rb_eArgError, "timeout should be positive");
             }
+
+            tv.tv_sec = NUM2INT(arg_timeout) / 1000000;
+            tv.tv_usec = NUM2INT(arg_timeout) % 1000000;
+            timeout = &tv;
         }
     } else {
         rb_raise(rb_eArgError, "invalid number of arguments");
     }
 
-    c = redisConnectNonBlock(StringValuePtr(arg_host), NUM2INT(arg_port));
+    redisOptions options = {0};
+    REDIS_OPTIONS_SET_TCP(&options, StringValuePtr(arg_host), NUM2INT(arg_port));
+    options.options |= REDIS_OPT_NONBLOCK;
+    options.connect_timeout = timeout;
+    c = redisConnectWithOptions(&options);
+
     return connection_generic_connect(self,c,arg_timeout);
 }
 
+static VALUE connection_connect_ssl(int argc, VALUE *argv, VALUE self) {
+    redisContext *c;
+    VALUE arg_host = Qnil;
+    VALUE arg_port = Qnil;
+    VALUE arg_timeout = Qnil;
+
+    struct timeval tv;
+    struct timeval *timeout = NULL;
+
+    redisSSLContext *ssl;
+    redisSSLContextError ssl_error;
+
+    if (argc == 2 || argc == 3) {
+        arg_host = argv[0];
+        arg_port = argv[1];
+
+        if (argc == 3) {
+            arg_timeout = argv[2];
+
+            /* Sanity check */
+            if (NUM2INT(arg_timeout) <= 0) {
+                rb_raise(rb_eArgError, "timeout should be positive");
+            }
+
+            tv.tv_sec = NUM2INT(arg_timeout) / 1000000;
+            tv.tv_usec = NUM2INT(arg_timeout) % 1000000;
+            timeout = &tv;
+        }
+    } else {
+        rb_raise(rb_eArgError, "invalid number of arguments");
+    }
+
+    redisInitOpenSSL();
+    ssl = redisCreateSSLContext(NULL, NULL, NULL, NULL, StringValuePtr(arg_host), &ssl_error);
+    SSL_CTX_set_verify(ssl->ssl_ctx, SSL_VERIFY_NONE, NULL);
```

Comment:
I don't think `SSL_VERIFY_NONE` should be the default. I'd say that the default should be verify is enabled, but you may pass a verify option.